### DISCUSSION TITLE
Проверка атрибута элемента

### DISCUSSION THREAD

**Discussion by DmitriyErmolov** [open-tutorials/cypress#26] [General]
Как с помощью cypress проверить, что после какого-либо действия в теге появился/удалился определенный атрибут, чтобы проверить валидацию js на стороне клиента?
![image](https://user-images.githubusercontent.com/97598347/216168018-4ca77e06-5065-4aa8-8a38-535e58060afc.png)
![image](https://user-images.githubusercontent.com/97598347/216168149-4338e748-a1f6-49ce-964e-aba2d58cbb6a.png)

**Comment by Vivioza5** [comment]
на одном из стендапов Антон упоминал о поиске атрибутов, попробуй посмотри в документации
[Assert the href attribute is equal to '/users'](https://docs.cypress.io/api/commands/should#Arguments)

```js
cy.get('#header a').should('have.attr', 'href', '/users')
```


в твоем случае как я понимаю 

```js
cy.get('#input').should('have.attr', 'aria-invalid', 'true')
```

или просто без конкретного значения атрибута 

```js
cy.get('#input').should('have.attr', 'aria-invalid')
```

**Reply by breslavsky** [reply]
Есть 2 варианта

```js
cy.get('selector').should('have.attr', 'aria-invalid').and('eq', 'true');
```

или

```js
cy.get('selector').invoke('attr', 'aria-invalid').should('eq', 'true');
```
